content/usgs-packages/geoknife_Discovery.Rmd

```diff
@@ -0,0 +1,233 @@
+
+Let's say that we were interested in evapotranspiration data. To search for which GDP datasets might contain evapotranspiration data, you can use the titles and abstracts.
+
+```{r grep-webdata, message=FALSE, warning=FALSE}
+# notice that you cannot perform a grep on all_webdata - it is because it is a special class
+# you need to perform pattern matching on vectors
+# grep("evapotranspiration", all_webdata)
+
+all_titles <- title(all_webdata)
+which_titles <- grep("evapotranspiration", all_titles)
+evap_titles <- all_titles[which_titles]
+head(evap_titles)
+
+all_abstracts <- abstract(all_webdata)
+which_abstracts <- grep("evapotranspiration", all_abstracts)
+evap_abstracts <- all_abstracts[which_abstracts]
+evap_abstracts[1]
+```
+
+`r length(unique(c(which_titles, which_abstracts)))` possible datasets to look through is a lot more manageable than `r length(all_webdata)`. Let's say the dataset titled "Yearly Conterminous U.S. actual evapotranspiration data" interested us enough to explore more. We have now identified a fabric of interest.
+
+We might want to know more about the dataset, such as what variables and time period is available. To actually create the fabric, you will need to use `webdata` and supply the appropriate datagroup object as the input. This should result in an object with a class of `webdata`. The following query and accessor functions will operate only on an object of class `webdata`.
+
+```{r create-fabric, message=FALSE, warning=FALSE}
+evap_fabric <- webdata(all_webdata["Yearly Conterminous U.S. actual evapotranspiration data"])
+class(evap_fabric)
+```
+
+Now that we have a defined fabric, we can explore what variables and time period are within that data. First, we use `query` to determine what variables exist. You'll notice that the accessor function `variable` returns NA. This is fine when you are just exploring available data; however, exploring avaialble times requires that the variable be defined. Thus, we need to set which variable from the dataset will be used. Then, we can explore times that are available in the data.
+
+```{r explore-fabric, message=FALSE, warning=FALSE}
+# no variables defined yet
+variables(evap_fabric)
+
+# find what variables are available
+query(evap_fabric, "variables")
+
+# trying to find available times before setting the variable results in an error
+# query(evap_fabric, "times")
+
+# only one variable, "et"
+variables(evap_fabric) <- "et"
+variables(evap_fabric)
+
+# now that the variable is set, we can explore available times
+query(evap_fabric, "times")
+```
+
+### Datasets not in GDP
+
+Any dataset available online that follows OPeNDAP protocol can be used with `geoknife`. These datasets can be found through web searches or other catalogs and require finding out the OPeNDAP endpoint (URL) for the dataset. This url is used as the input to the argument `url` in `webdata`. 
```

Comment:
Probably somewhere, it is basically the CF (climate and forecasting) conventions, plus some others that NetCDF-Java can handle nicely.  @dblodgett would have a better definition of what works and what doesn't.